### ISSUE TITLE
Enhance training progress logs: expose decoder loss and ETA

### ISSUE THREAD

**Issue by pablomoli** [pablomoli/niloc#30]
## Context

The PyTorch Lightning progress bar currently shows only `train_loss` (the summed encoder + decoder loss). Because `train_enc_loss_epoch` plateaus around ~10.2 very early and stays there by design (the encoder's job is coarse spatial priors), it dominates the visible total and masks decoder progress — which is the metric that actually matters for the scheduled-sampling regime past epoch ~150.

Concrete example from the 2026-04-12 fabrication-sprint retrain at epoch 285:

```
train_enc_loss_epoch  = 10.23   (plateaued — expected)
train_dec_loss_epoch  =  8.43   (still descending — what we care about)
train_loss_epoch      =  9.33   (what the progress bar prints)
```

At a glance the run looked stalled at 9.33 for 100+ epochs when the decoder was actually descending cleanly from 9.0 → 8.4. This forced us to manually dump the TensorBoard events file repeatedly just to confirm training health.

## Tasks

- [ ] Add a Lightning `Callback` (or extend the existing progress bar) that prints per-epoch:
  - `enc_loss` (epoch mean)
  - `dec_loss` (epoch mean) — **primary progress signal**
  - `total_loss` (epoch mean)
  - `tr_ratio`
  - `epoch_time_s`
- [ ] Track a rolling average of epoch duration (last 10 epochs) and print an **ETA to the configured `train_cfg.epochs`** each epoch, e.g. `ETA: 2h 47m (18.4 s/ep avg)`.
- [ ] Surface decoder loss in the checkpoint filename alongside the encoder loss so `ls` on the checkpoints dir is a useful progress view. Proposed format: `epoch=NNN-tr_ratio=0.XX-enc=10.23-dec=8.43.ckpt`.
- [ ] Keep the existing `train_enc_loss_epoch` scheduler monitor untouched — this is display-only.

## Acceptance

- A fresh run prints per-epoch `enc/dec/total/tr/ETA` to stdout in a single compact line.
- Checkpoint filenames include both encoder and decoder loss.
- No changes to training dynamics, scheduler monitor, or existing TensorBoard scalar names.

## Reference

- `niloc/trainer.py` — Lightning trainer setup, where the callback would be registered.
- `niloc/network/scheduled_2branch.py` — where `train_enc_loss_epoch` / `train_dec_loss_epoch` are logged.
- `train_synthetic.sh` — current entrypoint. Scheduler monitor is `train_enc_loss_epoch`; don't change.
